src/index.js

```diff
@@ -0,0 +1,362 @@
+  const rows = [];
+  dataKeys.forEach((dataRow) => {
+    const row = document.createElement('section');
+    row.className = 'row';
+
+    dataRow.forEach((dataKey) => {
+      const key = new Key(dataKey);
+      const button = key.generateKey();
+      row.append(button);
+    });
+    rows.push(row);
+  });
+  return rows;
+};
+const renderKeyBoard = () => {
+  const keyboard = document.createElement('section');
+  keyboard.className = 'keyboard';
+
+  const rows = renderRows(data);
+  rows.forEach((row) => {
+    keyboard.append(row);
+  });
+  return keyboard;
+};
+//---------------------------------------------
+// change Language
+const changeLanguage = (e) => {
+  e.preventDefault();
+  if (localStorage.getItem('language') === 'eng') {
+    localStorage.setItem('language', 'ru');
+  } else {
+    localStorage.setItem('language', 'eng');
+  }
+  document.querySelector('.keyboard').remove();
+  window.onload();
+};
+// mouse events..
+const mouseKeyUp = (e) => {
+  e.preventDefault();
+  const target = e.currentTarget;
+  let key = '';
+  if (!target) return;
+  target.childNodes.forEach((ch) => {
+    if (!ch.classList.contains('hide')) {
+      key = ch.textContent;
+    }
+  });
+  if (target.getAttribute('data-code') !== 'CapsLock') target.classList.remove('pressed');
+
+  if (target.classList.contains('button_special')) {
+    switch (key) {
+      case 'Shift':
+        if (localStorage.getItem('capslockPressed') !== 'false') {
+          pressShift();
+        } else {
+          unpressShift();
+        }
+        break;
+      case 'CapsLock':
+        if (localStorage.getItem('capslockPressed') === 'false') {
+          pressShift();
+          localStorage.setItem('capslockPressed', 'true');
+        } else {
+          unpressShift();
+          localStorage.setItem('capslockPressed', 'false');
+        }
+        break;
+      default:
+        break;
+    }
+  }
+};
+const mouseKeyOut = (e) => {
+  const target = e.currentTarget;
+  if (!target) return;
+  if (target.getAttribute('data-code') !== 'CapsLock') target.classList.remove('pressed');
+};
+const mouseKeyDown = (e) => {
+  e.preventDefault();
+  const target = e.currentTarget;
+  let key = '';
+  if (!target) return;
+  target.childNodes.forEach((ch) => {
+    if (!ch.classList.contains('hide')) {
+      key = ch.textContent;
+    }
+  });
+
+  if (target.getAttribute('data-code') !== 'CapsLock') {
+    target.classList.add('pressed');
+  } else {
+    target.classList.toggle('pressed');
+  }
+
+  const textarea = document.querySelector('#textarea');
+  if (!target.classList.contains('button_special')) {
+    let pozition = textarea.selectionStart;
+    const text = textarea.value;
+    const arr = text.split('');
+    arr.splice(pozition, 0, key);
+    textarea.value = arr.join('');
+    pozition += 1;
+    textarea.selectionStart = pozition;
+    textarea.selectionEnd = pozition;
+  } else {
+    switch (key) {
+      case 'Backspace': {
```

Comment:
As you see the logic for Backspace, Tab, Delete and Enter are almost the same. You can craete a separate function for the code inside case statement and depend on key value change splice and position values